R/merge_expression_module.R

```diff
@@ -335,7 +335,7 @@ merge_expression_srv <- function(id = "merge_id",
         merge_fun_name <- if (inherits(merge_function, "reactive")) merge_function() else merge_function
         check_merge_function(merge_fun_name)
 
-        ds <- Filter(Negate(is.null), lapply(selector_list(), function(x) x()))
+        ds <- Filter(function(e) length(e) > 0, lapply(selector_list(), function(x) x()))
```

Comment:
`selector_list` is an output from the data_extract_multiple_srv and it's a list of selectors and it's hard to imagine that [this line](https://github.com/insightsengineering/teal.transform/blob/8bff27130125046ca95915c1bb07c303449f0b55/R/data_extract_module.R#L619) wouldn't remove null's already. So NULL and length == 0 is not possible for this list.

Let's focus on the possible look of the selector. Each selector is a list which contains:
- filters <character>
- select <character>
- always selected <character>
- reshape <logical(1)>
- dataname <character(1)>
- internal_id <character(1)>
- keys <character>

To make sure that you fix [this](https://github.com/insightsengineering/teal.modules.general/blob/dad27779bcad9f0e83fa2a8ba8b24c5f6570949b/R/tm_outliers.R#L242) please fix it to the following way

```suggestion
        ds <- Filter(function(e) length(e$select) > 0, lapply(selector_list(), function(x) x()))
```


If we do so, we can close this issue:
- https://github.com/insightsengineering/teal.transform/issues/32

Reply:
This would require also:
- change the news file
- validate(need(length(ds) > 0, "Nothing selected"))

Reply:
Sorry, above suggestion and line in tmg where we removing selector with empty `-select` slot is wrong.
We need to overcome this in different way